ledger.go

```diff
@@ -809,13 +806,16 @@ func (l *Ledger) query() {
 		}(p)
 	}
 
-	wg.Wait()
-
-	votes := make([]*finalizationVote, 0, snowballK)
-	voters := make(map[AccountID]struct{}, snowballK)
+	votes := make([]*finalizationVote, 0, len(peers))
+	voters := make(map[AccountID]struct{}, len(peers))
 
 	for i := 0; i < cap(votes); i++ {
-		vote := <-voteChan
+		var vote *finalizationVote
+		select {
+		case vote = <-voteChan:
+		case <-time.After(conf.GetQueryTimeout()):
```

Comment:
i'd like this thing to be here at least for a while, because i reproduce it couple of times.
i have a feeling that since query function is running using metrics, metrics after being closed is not running the function and locks (either that was a race condition over metrics lock). nonetheless there was less than required results in vote channel.

Reply:
actually, thinking about that now, i think reason was in something else, maybe in having more peers than snowballK which resulted in channel being blocked on write inside goroutine.
currently i cannot reproduce it.